src/Val.h

```diff
@@ -234,6 +234,10 @@ class Val : public Obj {
 
     TableValPtr GetRecordFields();
 
+    // The version of the data generated by ToJSON and read by ValFromJSON. If that
+    // structure changes, this number must be increased.
+    static const int JSON_VERSION = 1;
+
```

Comment:
This doesn't feel great, would ditch the version.

Longer and IMO: We should just not change the output of `Val::ToJSON()` anymore (except for adding missing features or fixing obvious bugs). If there ever comes an idea for a drastically different JSON format, then that should be a new API like `Val::ToJSONv2()`, or implemented in a serializer outside of `Val.cc`.

Reply:
Seems reasonable to me.